File: 56-MergeInterval/merge.py
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        ret_arr=[]
        
        #O(nlog(n)) Time Complexity
        #O(n) Space complexity
        
        ret_arr.append(intervals[0])
        ret_count=0
        for i in range(1,len(intervals)):
            
            if ret_arr[ret_count][1]>=intervals[i][0]:
                ret_arr[ret_count][1]=max(intervals[i][1],ret_arr[ret_count][1])
            else:
                ret_arr.append(intervals[i])
                ret_count+=1
        #print(ret_arr)
        return ret_arr

File: 56-MergeInterval/merge.cpp
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
          vector<vector<int>> retVec;
  sort(intervals.begin(), intervals.end(),
          [](const std::vector<int>& a, const std::vector<int>& b) {
    return a[0] < b[0];
  });
  
  
  int start = -1, end = -1;
  start=intervals[0][0];
  end=intervals[0][1];


  for (int i = 0; i < intervals.size(); i++) {
      auto currInterval = intervals[i];
      if (end>=currInterval[0])
        end = max(end, currInterval[1]);
      else {
        retVec.push_back({start, end});
        start = currInterval[0];
        end = currInterval[1];
      }
  }
  retVec.push_back({start, end});
  return retVec;      
  
}